### PR TITLE
feat: Support for ES 7, ES field names containing dots, ES field sort order

### PR DESCRIPTION
**Pull request by jaredbwasserman**
### Summary of Changes
1. Elasticsearch base extractor now able to retrieve mapping properties for Elasticsearch >= 7.x. This update is needed because the mapping format was updated following the removal of types (https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html).
2. Field names containing dots now correctly extracted in Elasticsearch metadata extractor.
3. Field sort order now optionally corrected in Elasticsearch metadata extractor.
4. This PR is created because the previous one was closed: https://github.com/amundsen-io/amundsen/pull/1568

### Tests
1. Modified `TestElasticsearchColumnStatsExtractor.test_extractor_without_technical_data_v6` (previously called `TestElasticsearchColumnStatsExtractor._test_extractor_without_technical_data`) to explicitly test Elasticsearch version 6 and lower.
2. Added `TestElasticsearchColumnStatsExtractor.test_extractor_without_technical_data_v7` which is the same test as `TestElasticsearchColumnStatsExtractor.test_extractor_without_technical_data_v6` but for Elasticsearch version 7 and higher.
3. Modified `TestElasticsearchIndexExtractor.test_extractor_without_technical_data_es_v6` (previously called `TestElasticsearchIndexExtractor.test_extractor_without_technical_data) to explicitly test Elasticsearch version 6 and lower.
4. Added `TestElasticsearchIndexExtractor.test_extractor_without_technical_data_es_v7` which is the same test as `TestElasticsearchIndexExtractor.test_extractor_without_technical_data_es_v6` but for Elasticsearch version 7 and higher.
5. Added `TestElasticsearchIndexExtractor.test_extractor_hierarchical_es_v6` to test field names with dots are extracted correctly for Elasticsearch version 6 and lower.
6. Added `TestElasticsearchIndexExtractor.test_extractor_hierarchical_es_v7` to test field names with dots are extracted correctly for Elasticsearch version 7 and higher.
7. Added `TestElasticsearchIndexExtractor.test_extractor_sorted_es_v6` to test field sort order is correct for Elasticsearch version 6 and lower.
8. Added `TestElasticsearchIndexExtractor.test_extractor_sorted_es_v7` to test field sort order is correct for Elasticsearch version 7 and higher.

### Documentation

Added information in the [README](https://github.com/amundsen-io/amundsen/blob/main/databuilder/README.md#elasticsearchmetadataextractor) for `ElasticsearchMetadataExtractor` discussing the newly added `correct_sort_order` conf option.

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: databuilder/databuilder/extractor/es_base_extractor.py [hunk 1]
```diff
@@ -47,13 +58,43 @@ def _get_indexes(self) -> Dict:
...
+        cols: List[ColumnMetadata] = []
+
+        for col_name, col_mapping in input_mapping.items():
+            qualified_col_name = str(parent_col_name) + separator + col_name if parent_col_name else col_name
+            if isinstance(col_mapping, dict):
+                if col_mapping.__contains__('properties'):
+                    # Need to recurse
+                    inner_mapping: Dict = col_mapping.get('properties', {})
+                    cols.extend(self._get_attributes(input_mapping=inner_mapping,
+                                                     parent_col_name=qualified_col_name,
```
Comment (mgorsk1) [inline databuilder/databuilder/extractor/es_base_extractor.py:88]: signature of _get_attributes expects boolean value in parent_col_name but you pass string here. can you correct the signature?

**Review by mgorsk1 — COMMENTED** [review]
lgtm, one nit

File: databuilder/databuilder/extractor/es_base_extractor.py (hunk 1 above)
Comment (jaredbwasserman) [reply databuilder/databuilder/extractor/es_base_extractor.py:88]: Good catch. I will fix this.

**Comment by jaredbwasserman** [comment]
Thanks @mgorsk1 - I fixed the function signature, fixed the issue found by mypy check, and bumped the databuilder version.

File: databuilder/tests/unit/extractor/test_es_column_stats_extractor.py [hunk 2]
```diff
@@ -87,11 +145,12 @@ def _get_extractor(self) -> Any:
 
         return extractor
 
-    def test_extractor_without_technical_data(self) -> None:
+    def _test_extractor_without_technical_data(self, es_version: str, indices: Dict, stats: Dict) -> None:
```
Comment (mgorsk1) [inline databuilder/tests/unit/extractor/test_es_column_stats_extractor.py:148]: maybe this should have different name now?

Comment (jaredbwasserman) [reply databuilder/tests/unit/extractor/test_es_column_stats_extractor.py:148]: Good point. What do you think of `_extract_without_technical_data`?

**Review by mgorsk1 — APPROVED** [review]

**Comment by mgorsk1** [comment]
thanks @jaredbwasserman , nice work!

**Pull request merged**